protected/humhub/components/behaviors/PolymorphicRelation.php

```diff
@@ -92,28 +101,53 @@ public function setPolymorphicRelation($object)
     {
         if ($this->validateUnderlyingObjectType($object)) {
             $this->cached = $object;
-            if ($object instanceof ActiveRecord) {
+            if ($object instanceof ActiveRecordInterface) {
                 $this->owner->setAttribute($this->classAttribute, self::getObjectModel($object));
                 $this->owner->setAttribute($this->pkAttribute, $object->getPrimaryKey());
             }
+
+            /**
+             * @since 1.16
+             */
+            if ($object->hasMethod('onSetPolymorphicRelationOwner')) {
```

Comment:
- Couldn't we use `PolymorphicRelation::onSet` instead of these two `hasMethod()` checks?

- As alternative, we may could add an option `PolymorphicRelation::harmonizeGuid` which makes sure the guid of both are identical and set.

Reply:
Yes, that would be an option and I think that was my original idea.

Maybe it would make sense to define the `$onSet` field as an array of callbacks:

```php
/**
 * @var callable[]
 * @since 1.16
 */
public array $onSet = [];
```

and the call back would have a uniform signature:

```php
public function(PolymorphicRelation $relation, ActiveRecordInterface $owner, object $object): void;
// or more restricted
public function(PolymorphicRelation $relation, ActiveRecordInterface $owner, ActiveRecordInterface $object): void;
```

When calling the callback, the latter is given in array notation, and the first element is a class name that corresponds to the class of the $object, then the method from the second element will be called on this particular `$object`, and not statically.